### PR TITLE
Clean up main method a little

### PR DESCRIPTION
**Pull request by pbrinkmeier**
This PR replaces the old giant iflist by a dispatcher class that takes care of running the right command when the corresponding flag is set.

### CODE REVIEW

File: src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import com.github.firmwehr.gentle.util.Pair;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class CommandDispatcher {
+	// TODO: Not a very pretty type. Perhaps this could be its own record.
```
Comment (I-Al-Istannen) [inline src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java:13]: I'd appreciate it

File: src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		List<Consumer<Path>> commands =
+			this.commands.stream().filter(cmd -> cmd.first().apply(arguments)).map(Pair::second).toList();
+
+		if (commands.size() == 0) {
+			UserOutput.userError("No operation specified.");
+			System.exit(1);
+		}
+
+		if (commands.size() > 1) {
+			// FIXME: Print conflicting flags.
```
Comment (I-Al-Istannen) [inline src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java:36]: I don't think that is easily possible with your model as you do not have any information about the name of the flag. You'd need access to the `java.lang.reflect.Method` or sth like it.

Or you pass it in a parameter and store that in the record as well, which is probably a good compromise.

File: src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		this.commands.add(new Pair<>(flag, command));
+		return this;
+	}
+
+	public void dispatch(String[] args) {
+		CommandArguments arguments = new CommandArgumentsParser().parseOrExit(args);
+		List<Consumer<Path>> commands =
+			this.commands.stream().filter(cmd -> cmd.first().apply(arguments)).map(Pair::second).toList();
+
+		if (commands.size() == 0) {
```
Comment (I-Al-Istannen) [inline src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java:30]:
```suggestion
		if (commands.isEmpty()) {
```

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java (hunk 2 above)
Comment (pbrinkmeier) [reply src/main/java/com/github/firmwehr/gentle/cli/CommandDispatcher.java:36]: Apparently there isn't really a neat way to access the `Method` instance when a method is used as a function. For now we'll just have to cope with a bit of redundancy.

**Comment by pbrinkmeier** [comment]
Not sure if we really need that last commit. I think I'd prefer to have all those string literals in a single place but it's not a big concern at all.

**Comment by pbrinkmeier** [comment]
I guess the consistent way would be to just move to the Command Pattern completely.

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**